boxsdk/object/file.py

```diff
@@ -275,7 +280,14 @@ def update_contents_with_stream(
 
         data = {'attributes': json.dumps(attributes)}
         files = {'file': ('unused', file_stream)}
-        headers = {'If-Match': etag} if etag is not None else None
+        headers = {}
+        if etag is not None:
+            headers['If-Match'] = etag
+        if sha1 is not None:
+            # The Content-MD5 field accepts sha1
+            headers['Content-MD5'] = sha1
+        if not headers:
+            headers = None
```

Comment:
If we make this into a helper method, we could replace these two lines with `return headers or None`

Reply:
We feel that a helper method for just three instances of creating headers may be less readable than simply including the header logic multiple times. Also, setting the headers in the method is how it is currently handled in other methods in the SDK.

Reply:
Was thinking it could be nice to share this for file/folder. If we have more headers to add in the future for upload we then have a shared place to update.

Yea it's not a big deal overall so either way is okay, thanks for considering it!